### PR TITLE
chore(updatedependencies): update various dependencies

### PR DESCRIPTION
**Pull request by dogpatch626**
this is a pull request to update various dependencies:
```
"chalk": "^3.0.0",
"@semantic-release/changelog": "^6.0.0",
    "@semantic-release/commit-analyzer": "^9.0.0",
    "@semantic-release/github": "^8.0.0",
    "@semantic-release/github": "^8.0.0",
    "@semantic-release/npm": "^9.0.0",
    "@semantic-release/release-notes-generator": "^10.0.0",
    "jest": "^29.0.0",
       "@babel/cli": "^7.0.0",
    "babel-preset-amex": "^4.0.0",
    "semantic-release": "^19.0.0"
```
## How Has This Been Tested?
This was tested through unit tests and running npm pack to install in one app.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [x] Dependency update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] My changes are in sync with the code style of this project.
- [ ] There aren't any other open Pull Requests for the same issue/update.
- [ ] These changes should be applied to a maintenance branch.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: __tests__/bin/utils/setMiddlewareUtils.spec.js [hunk 1]
```diff
@@ -174,7 +176,9 @@ describe('Set Middleware Utils', () => {
       }
       expect(mockFs.linkSync).toHaveBeenCalled();
       expect(mockFs.symlinkSync).not.toHaveBeenCalled();
-      process.platform = originalPlatform;
+      Object.defineProperty(process, 'platform', {
+        value: originalPlatform,
+      });
```
Comment (Matthew-Mallimo) [inline __tests__/bin/utils/setMiddlewareUtils.spec.js:181]: why was this necessary?

File: __tests__/src/middleware.spec.js [hunk 2]
```diff
@@ -49,12 +49,15 @@ describe('pipe', () => {
   beforeEach(() => {
     request.__resetRequests(); // eslint-disable-line no-underscore-dangle
     request.mockClear();
+    // jest.resetAllMocks();
```
Comment (Matthew-Mallimo) [inline __tests__/src/middleware.spec.js:52]: Let's remove commented code

File: package.json [hunk 3]
```diff
@@ -101,7 +101,7 @@
       "@semantic-release/release-notes-generator",
       "@semantic-release/changelog",
       "@semantic-release/npm",
-      "@semantic-release/git",
+      "@semantic-release/github",
```
Comment (Matthew-Mallimo) [inline package.json:104]: Looks like this plugin is already there, one line below

**Review by Matthew-Mallimo — CHANGES_REQUESTED** [review]

File: __tests__/bin/utils/setMiddlewareUtils.spec.js (hunk 1 above)
Comment (dogpatch626) [reply __tests__/bin/utils/setMiddlewareUtils.spec.js:181]: So the way this was written before ```process.platform = originaPlatform``` would make tests fail because I believe that you can't directly change that value Im pretty sure it's read only but you can do it in this way to change the value of process.platform. the test itself changes it to win32 and then back to the original platform.

File: __tests__/src/middleware.spec.js [hunk 4]
```diff
@@ -70,7 +71,7 @@ describe('pipe', () => {
   });
 
   it('should show a message when there is a socket error', () => {
-    jest.spyOn(console, 'error');
+    console.error = jest.fn();
```
Comment (10xLaCroixDrinker) [inline __tests__/src/middleware.spec.js:74]: why this change?

Comment (dogpatch626) [reply __tests__/src/middleware.spec.js:74]: There was an issue where jest would pick up the console.error and report it back as a test failure? even though we spy on console.error. so Matt suggested trying that out and it worked fine including snapshots for it.

Comment (10xLaCroixDrinker) [inline __tests__/src/middleware.spec.js:74]:
```suggestion
    jest.spyOn(console, 'error').mockImplementation(() => 0);
```

**Review by 10xLaCroixDrinker — CHANGES_REQUESTED** [review]

**Review by giulianok — APPROVED** [review]

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Review by Matthew-Mallimo — APPROVED** [review]
